src/Features/LanguageServer/Protocol/Features/Diagnostics/DiagnosticAnalyzerService.cs

```diff
@@ -110,9 +111,10 @@ public void RequestDiagnosticRefresh()
             priorityProvider ??= new DefaultCodeActionRequestPriorityProvider();
 
             // always make sure that analyzer is called on background thread.
-            return Task.Run(() => analyzer.GetDiagnosticsForSpanAsync(
+            await TaskScheduler.Default;
```

Comment:
📝 This is slightly different from before. If the caller is already on the background thread, it will no longer yield. You can use `SwitchTo(true)` to force it to yield if that behavior is important.

Reply:
this was intentional.  as the comment says, this is just to ensure that we're on the BG.  if we're already there, no need to yield.